libsolidity/analysis/TypeChecker.cpp

```diff
@@ -645,14 +645,23 @@ bool TypeChecker::visit(VariableDeclaration const& _variable)
 			if (!allowed)
 				m_errorReporter.typeError(_variable.location(), "Constants of non-value type not yet implemented.");
 		}
+
 		if (!_variable.value())
 			m_errorReporter.typeError(_variable.location(), "Uninitialized \"constant\" variable.");
 		else if (!_variable.value()->annotation().isPure)
-			m_errorReporter.warning(
-				_variable.value()->location(),
-				"Initial value for constant variable has to be compile-time constant. "
-				"This will fail to compile with the next breaking version change."
-			);
+		{
+			if (m_scope->sourceUnit().annotation().experimentalFeatures.count(ExperimentalFeature::V050))
```

Comment:
I think it is worth considering adding a `v050` bool on the `TypeChecker` context since it is going to be used in a couple of places.

Here `m_scope` shouldn't be null given there should be no way to reach a `VariableDeclaration` outside a `ContractDefinition`, but it could be `nullptr` otherwise. We usually add assertions for these so the sensible way is to check if `m_scope` is not null before checking the rest.

Reply:
Wait a second, `VariableDeclaration` should have a `sourceUnit()` method too.

Reply:
+1 to adding a `v050` bool. I'll check the `sourceUnit()` in the VariableDeclaration.